keychain-sdk/config.go

```diff
@@ -58,3 +49,25 @@ type Config struct {
 	// failed (but the blockchain might still include in a block later).
 	TxTimeout time.Duration
 }
+
+// Config is the configuration for the Keychain.
+type Config struct {
+	BasicConfig
+
+	// ConsensusNodeThreshold represents the number of nodes required to execute a pending key/sign request.
+	ConsensusNodeThreshold uint8
+
+	// GRPCURLs is the list of URLs of the gRPC server to connect to.
+	// e.g. "localhost:9090"
+	GRPCConfigs []GrpcNodeConfig
+}
+
+type GrpcNodeConfig struct {
+	// GRPCInsecure determines whether to allow an insecure connection to the
+	// gRPC server.
+	GRPCInsecure bool
+
+	// GRPCURL is the URL of the gRPC server to connect to.
+	// e.g. "localhost:9090"
+	GRPCURL string
+}
```

Comment:
since we are inside a struct called "GrpcNodeConfig", let's avoid repeating the word GRPC everywhere:

```suggestion
type GrpcNodeConfig struct {
	// Insecure determines whether to allow an insecure connection to the
	// gRPC server.
	Insecure bool

	// Host is the address of the gRPC server to connect to.
	// e.g. "localhost:9090"
	Host string
}
```

technically, the second field is not a URL since it doesn't contain a protocol scheme, i renamed it into "host"